include/argparse/argparse.hpp

```diff
@@ -640,15 +640,15 @@ class Argument {
     };
 
     // precondition: we have consumed or will consume at least one digit
-    auto consume_digits = [=](std::string_view s) {
-      auto it = std::find_if_not(std::begin(s), std::end(s), is_digit);
-      return s.substr(it - std::begin(s));
+    auto consume_digits = [=](std::string_view needle) {
+      auto it = std::find_if_not(std::begin(needle), std::end(needle), is_digit);
+      return needle.substr(static_cast<std::size_t>(it - std::begin(needle)));
```

Comment:
I don't see a way around this cast, but is this the right type? For containers, I thought we should use `size_type`, e.g. `static_cast<decltype(needle)::size_type>(it - std::begin(needle))`.

Yes, these all should resolve to the same type (probably `std::size_t`), but this may save us if the type for `needle` changes in the future.

Reply:
Yes for sure, it will save in future